src/ast/variables/ExportDefaultVariable.ts

```diff
@@ -23,9 +23,11 @@ export default class ExportDefaultVariable extends LocalVariable {
 	}
 
 	addReference(identifier: Identifier) {
-		this.name = identifier.name;
-		if (this.original !== null) {
-			this.original.addReference(identifier);
+		if (!this.hasId) {
+			this.name = identifier.name;
+			if (this.original !== null) {
+				this.original.addReference(identifier);
+			}
```

Comment:
Seems like a good approach! The original logic was probably only in place to provide a better name than the file name to unnamed default exports. If we have an id, using that makes a lot of sense.